imapserver/capability.go

```diff
@@ -86,6 +88,21 @@ func (c *Conn) availableCaps() []imap.Cap {
 
 		// Capabilities which require backend support and apply to both
 		// IMAP4rev1 and IMAP4rev2
+		if appendLimitSession, ok := c.session.(SessionAppendLimit); ok {
+			if appendLimitSession.DiscloseLimit() {
+				limit := appendLimitSession.AppendLimit()
+				caps = append(caps, imap.Cap(fmt.Sprintf("APPENDLIMIT=%d", limit)))
+			} else {
+				caps = append(caps, imap.CapAppendLimit)
+			}
+		} else if limit, ok := available.AppendLimit(); ok {
```

Comment:
This adds two ways to set the global limit, either with a `SessionAppendLimit`, either with manual string formatting in `Caps`. I'd prefer to keep only the former.